servers/quarkus-cli/src/main/java/org/projectnessie/quarkus/cli/CheckContent.java

```diff
@@ -0,0 +1,264 @@
+import org.projectnessie.versioned.Key;
+import org.projectnessie.versioned.ReferenceInfo;
+import org.projectnessie.versioned.ReferenceNotFoundException;
+import org.projectnessie.versioned.StoreWorker;
+import org.projectnessie.versioned.persist.adapter.ContentAndState;
+import org.projectnessie.versioned.persist.adapter.KeyFilterPredicate;
+import org.projectnessie.versioned.persist.adapter.KeyListEntry;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "check-content",
+    mixinStandardHelpOptions = true,
+    description = "Check content readability of active keys.")
+public class CheckContent extends BaseCommand {
+
+  @CommandLine.Option(
+      names = {"-o", "--output"},
+      description =
+          "JSON output file name or '-' for STDOUT. If not set, per-key status is not reported.")
+  private String outputSpec;
+
+  @CommandLine.Option(
+      names = {"-k", "--key-element"},
+      description =
+          "Elements or a specific content key to check (zero or more). If not set, all current keys will be checked.")
+  private List<String> keyElements;
+
+  @CommandLine.Option(
+      names = {"-c", "--show-content"},
+      description = "Include content for each valid key in the output.")
+  private boolean showContent;
+
+  @CommandLine.Option(
+      names = {"-B", "--batch"},
+      defaultValue = "25",
+      description = "The max number of keys to load at the same time.")
+  private int batchSize;
+
+  @CommandLine.Option(
+      names = {"-W", "--worker-class"},
+      defaultValue = "org.projectnessie.server.store.TableCommitMetaStoreWorker",
+      description = "The class name of the Nessie Store Worker (internal).")
+  private String workerClass;
+
+  @CommandLine.Option(
+      names = {"-r", "--ref"},
+      description = "Reference name to use (default branch, if not set).")
+  private String ref;
+
+  @CommandLine.Option(
+      names = {"-H", "--hash"},
+      description = "Commit hash to use (defaults to the HEAD of the specified reference).")
+  private String hash;
+
+  @CommandLine.Option(
+      names = {"-s", "--summary"},
+      description = "Print a summary of results to STDOUT (irrespective of the --output option).")
+  private boolean summary;
+
+  @CommandLine.Option(
+      names = {"-E", "--error-only"},
+      description = "Produce JSON only for keys with errors.")
+  private boolean errorOnly;
+
+  private final AtomicInteger keysProcessed = new AtomicInteger();
+  private final AtomicInteger errorDetected = new AtomicInteger();
+
+  @Override
+  public Integer call() throws Exception {
+    warnOnInMemory();
+
+    if (outputSpec != null) {
+      if ("-".equals(outputSpec)) {
+        check(spec.commandLine().getOut());
+        spec.commandLine().getOut().println();
+      } else {
+        try (PrintWriter out = new PrintWriter(outputSpec)) {
+          check(out);
+        }
+      }
+    } else {
+      check(new PrintWriter(NullOutputStream.NULL_OUTPUT_STREAM));
```

Comment:
Can you add a charset to the PrintWriter constructor?

Reply:
added